File: loop_transformations/sources/apply_syntax_equivalent.2.f90
! @@name:       apply_syntax_equivalent.2
! @@type:       F-free
! @@operation:  compile
! @@expect:     success
! @@version:    omp_6.0
subroutine equivalent(A)
   implicit none
   double precision :: A(0:2,   0:99,   0:99)
   integer          ::   k,    j1,j2,  i1,i2

   !$omp interchange       !! grid modifier
   do i1 = 0, 24
   !$omp nothing           !! grid modifier
   do j1 = 0, 24

      !$omp nothing        !! intratile modifier
      do i2 = 0, 3
      !$omp interchange    !! intratile modifier
      do j2 = 0, 3

         do k = 0, 2
            i = i1 * 4 + i2
            j = j1 * 4 + j2
            A(k,j,i) = A(k,j,i) + 1
         enddo

      enddo; enddo
   enddo; enddo
end subroutine

File: loop_transformations/sources/apply_syntax_equivalent.2.c
/*
* @@name:   apply_syntax_equivalent.2
* @@type:   C
* @@operation: compile
* @@expect: success
* @@version:   omp_6.0
*/
void equivalent(double A[100][100][3])
{
  #pragma omp interchange
  for (int i1 = 0; i1 < 25; ++i1)
  #pragma omp nothing
  for (int j1 = 0; j1 < 25; ++j1)

     #pragma omp nothing
     for (int i2 = 0; i2 < 4; ++i2)
     #pragma omp interchange
     for (int j2 = 0; j2 < 4; ++j2)

        for (int k = 0; k < 3; ++k) {
           int i = i1 * 4 + i2;
           int j = j1 * 4 + j2;
           A[i][j][k] = A[i][j][k] + 1;
        }
}
